store/sqlstore/file_info_store.go

```diff
@@ -95,6 +96,72 @@ func (fs SqlFileInfoStore) Get(id string) (*model.FileInfo, *model.AppError) {
 	return info, nil
 }
 
+func (fs SqlFileInfoStore) GetWithOptions(opt *model.GetFilesOptions) ([]*model.FileInfo, *model.AppError) {
+	if opt == nil {
+		opt = &model.GetFilesOptions{}
+	}
+
+	query := fs.getQueryBuilder().
+		Select("FileInfo.*").
+		From("FileInfo").
+		LeftJoin("Posts ON FileInfo.PostId = Posts.Id") // Left join to get files that aren't attached to a post
+
+	if len(opt.ChannelIds) > 0 {
+		query = query.Where(sq.Eq{"Posts.ChannelId": opt.ChannelIds})
+	}
+
+	if len(opt.UserIds) > 0 {
+		query = query.Where(sq.Eq{"FileInfo.CreatorId": opt.UserIds})
+	}
+
+	if opt.Since > 0 {
+		query = query.Where(sq.Gt{"FileInfo.CreateAt": opt.Since})
+	}
+
+	if !opt.IncludeDeleted {
+		query = query.Where("FileInfo.DeleteAt = 0")
+	}
+
+	sortDirection := model.FILE_SORT_ORDER_ASCENDING
+	if opt.SortDirection != "" {
+		sortDirection = opt.SortDirection
+	}
+	if opt.SortBy != "" {
+		switch opt.SortBy {
+		case model.FILE_SORT_BY_CREATED, model.FILE_SORT_BY_SIZE:
+			query = query.OrderBy(fmt.Sprintf("FileInfo.%s %s", opt.SortBy, sortDirection))
```

Comment:
Let's make all sorts stable by adding a secondary sort on `Id` for when two file infos have the same property as the primary sort.

Reply:
Also, can we write out the `OrderBy` explicitly for these two fields instead of using `fmt.Sprintf`? I think the code would be clearer as a result.